File: sample/CStyleSamplesVB/Samples/SkinDetector.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports OpenCvSharp
Imports OpenCvSharp.CPlusPlus

' Namespace OpenCvSharpSamplesVB
Imports SampleBase

''' <summary>
''' CvAdaptiveSkinDetector sample
''' </summary>
    Friend Module SkinDetector
        ''' <summary>
        ''' 
        ''' </summary>
        Public Sub Start()
        Using imgSrc As New IplImage(FilePath.Image.Balloon, LoadMode.Color)
            Using imgHueMask As New IplImage(imgSrc.Size, BitDepth.U8, 1)
                Using imgDst As IplImage = imgSrc.Clone()
                    Dim detector As New CvAdaptiveSkinDetector(1, MorphingMethod.None)
                    detector.Process(imgSrc, imgHueMask)
                    DisplaySkinPoints(imgHueMask, imgDst, CvColor.Green)

                    Using windowSrc As New CvWindow("src", imgSrc)
                        Using windowDst As New CvWindow("skin", imgDst)
                            Cv.WaitKey(0)
                        End Using
                    End Using
                End Using
            End Using
        End Using
        End Sub

        ''' <summary>
        ''' 
        ''' </summary>
        ''' <param name="imgHueMask"></param>
        ''' <param name="imgRgbDst"></param>
        ''' <param name="color"></param>
        Private Sub DisplaySkinPoints(ByVal imgHueMask As IplImage, ByVal imgRgbDst As IplImage, ByVal color As CvColor)
            If imgHueMask.Size <> imgRgbDst.Size Then
                Throw New ArgumentException()
            End If

            For y As Integer = 0 To imgHueMask.Height - 1
                For x As Integer = 0 To imgHueMask.Width - 1
                    Dim value As Byte = CByte(imgHueMask(y, x).Val0)
                    If value <> 0 Then
                        imgRgbDst(y, x) = color
                    End If
                Next x
            Next y
        End Sub
    End Module
' End Namespace

File: sample/CStyleSamplesCS/Samples/SkinDetector.cs
using System;
using OpenCvSharp;
using OpenCvSharp.CPlusPlus;
using SampleBase;

namespace CStyleSamplesCS
{
    /// <summary>
    /// CvAdaptiveSkinDetector sample
    /// </summary>
    class SkinDetector
    {
        /// <summary>
        /// 
        /// </summary>
        public SkinDetector()
        {
            using (IplImage imgSrc = new IplImage(FilePath.Image.Balloon, LoadMode.Color))            
            using (IplImage imgHueMask = new IplImage(imgSrc.Size, BitDepth.U8, 1))
            using (IplImage imgDst = imgSrc.Clone())
            {
                CvAdaptiveSkinDetector detector = new CvAdaptiveSkinDetector(1, MorphingMethod.None);
                detector.Process(imgSrc, imgHueMask);
                DisplaySkinPoints(imgHueMask, imgDst, CvColor.Green);

                using (new CvWindow("src", imgSrc))
                using (new CvWindow("skin", imgDst))
                {
                    Cv.WaitKey(0);
                }
            }
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="imgHueMask"></param>
        /// <param name="imgRgbDst"></param>
        /// <param name="color"></param>
        private void DisplaySkinPoints(IplImage imgHueMask, IplImage imgRgbDst, CvColor color)
        {
            if (imgHueMask.Size != imgRgbDst.Size)
                throw new ArgumentException();

            for (int y = 0; y < imgHueMask.Height; y++)
            {
                for (int x = 0; x < imgHueMask.Width; x++)
                {
                    byte value = (byte)imgHueMask[y, x].Val0;
                    if (value != 0)
                    {
                        imgRgbDst[y, x] = color;
                    }
                }
            }
        }
    }
}